app/connectors/getFinancialDetails/FinancialDetailsConnector.scala

```diff
@@ -0,0 +1,98 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors.getFinancialDetails
+
+import config.AppConfig
+import connectors.parsers.getFinancialDetails.FinancialDetailsParser.{GetFinancialDetailsFailureResponse, GetFinancialDetailsResponse}
+import models.EnrolmentKey
+import models.EnrolmentKey._
+import models.TaxRegime.{ITSA, VAT}
+import play.api.http.Status.INTERNAL_SERVER_ERROR
+import uk.gov.hmrc.http.HttpReads.Implicits._
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse, UpstreamErrorResponse}
+import utils.Logger.logger
+import utils.PagerDutyHelper
+import utils.PagerDutyHelper.PagerDutyKeys._
+
+import java.util.UUID.randomUUID
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class FinancialDetailsConnector @Inject()(httpClient: HttpClient,
+                                          appConfig: AppConfig)
+                                         (implicit ec: ExecutionContext) {
+
+  private def headers: Seq[(String, String)] = Seq(
+    "Authorization" -> s"Bearer ${appConfig.eiOutboundBearerToken}",
+    "CorrelationId" -> randomUUID().toString,
+    "Environment" -> appConfig.eisEnvironment
+  )
+
+  def getFinancialDetails(enrolmentKey: EnrolmentKey, overridingParameters: Option[String])(implicit hc: HeaderCarrier): Future[GetFinancialDetailsResponse] = {
+    val parameters = overridingParameters.fold(appConfig.queryParametersForGetFinancialDetails + appConfig.addDateRangeQueryParameters())(_ + appConfig.addDateRangeQueryParameters())
+    httpClient.GET[GetFinancialDetailsResponse](url =
+      appConfig.getRegimeFinancialDetailsUrl(enrolmentKey) + parameters, headers = headers).recover {
+      case e: UpstreamErrorResponse => {
+        PagerDutyHelper.logStatusCode("getFinancialDetails", e.statusCode)(RECEIVED_4XX_FROM_1811_API, RECEIVED_5XX_FROM_1811_API)
+        logger.error(s"[FinancialDetailsConnector][getFinancialDetails] - Received ${e.statusCode} status from API 1811 call - returning status to caller")
+        Left(GetFinancialDetailsFailureResponse(e.statusCode))
+      }
+      case e: Exception => {
+        PagerDutyHelper.log("getFinancialDetails", UNKNOWN_EXCEPTION_CALLING_1811_API)
+        logger.error(s"[FinancialDetailsConnector][getFinancialDetails] - An unknown exception occurred - returning 500 back to caller - message: ${e.getMessage}")
+        Left(GetFinancialDetailsFailureResponse(INTERNAL_SERVER_ERROR))
+      }
+    }
+  }
+
+  private def getFinancialDetailsUrl(enrolmentKey: EnrolmentKey) = enrolmentKey match {
+    case EnrolmentKey(VAT, VRN, vrn) => appConfig.getVatFinancialDetailsUrl(vrn)
+    case EnrolmentKey(ITSA, NINO, utr) => appConfig.getItsaFinancialDetailsUrl(utr)
+    //case EnrolmentKey(CT, UTR, utr) => appConfig.getCtFinancialDetailsUrl(utr)
```

Comment:
This shows that we are not making the API generic (as the CR asks for) and would require more changes to onboard a new service in the future. We should be capturing the regimeType, idType etc from the API consumer and passing it around in the EnrolmentKey, then we don't need functions like this at all